### PR TITLE
Output user-friendly warning message for conditional effect errors.

### PR DESCRIPTION
**Pull request by gabryon99**

### CODE REVIEW

File: plugins/formal-verification/formver.plugin/src/org/jetbrains/kotlin/formver/reporting/VerifierErrorInterpreter.kt [hunk 1]
```diff
@@ -99,4 +106,40 @@ class VerifierErrorInterpreter {
             is SourceRole.ReturnsEffect.Null -> if (negated) "null" else "non-null"
             else -> TODO("Unreachable")
         }
+
+    private fun SourceRole.ReturnsEffect.describe(): String = when (this) {
+        is SourceRole.ReturnsEffect.Bool -> "a $bool value is returned"
+        is SourceRole.ReturnsEffect.Null -> "a ${if (negated) "non-null" else "null"} value is returned"
+        SourceRole.ReturnsEffect.Wildcard -> "the function returns"
+    }
```
Comment (jesyspa) [inline plugins/formal-verification/formver.plugin/src/org/jetbrains/kotlin/formver/reporting/VerifierErrorInterpreter.kt:114]: Can this be unified with the `ReturnsEffect` logic?

File: plugins/formal-verification/testData/diagnostics/bad_contracts/cond_effects.fir.diag.txt [hunk 2]
```diff
@@ -11,9 +21,9 @@ method global$fun_mayReturnNonNull$fun_take$NT_Any$return$NT_Any(local$x: dom$Nu
   label label$ret$0
 }
 
-/cond_effects.kt:(178,210): warning: Viper verification error: Postcondition of global$fun_mayReturnNonNull$fun_take$NT_Any$return$NT_Any might not hold. Assertion ret$0 == (dom$Nullable$null(): dom$Nullable[dom$Any]) ==> dom$Type$isSubtype((dom$TypeOf$typeOf(local$x): dom$Type), dom$Type$Int()) might not hold.
+/cond_effects.kt:(328,360): warning: Cannot verify that if a null value is returned then 'x is kotlin.Int'.
```
Comment (jesyspa) [inline plugins/formal-verification/testData/diagnostics/bad_contracts/cond_effects.fir.diag.txt:24]: It seems a bit distracting to use the fully qualified type here, WDYT?

File: plugins/formal-verification/formver.plugin/src/org/jetbrains/kotlin/formver/FormalVerificationPluginErrorMessages.kt [hunk 3]
```diff
@@ -55,5 +55,11 @@ object FormalVerificationPluginErrorMessages : BaseDiagnosticRendererFactory() {
             "Function ''{0}'' may leak from its in-place contract.",
             FirDiagnosticRenderers.SYMBOL,
         )
+        put(
+            PluginErrors.CONDITIONAL_EFFECT_ERROR,
+            "Cannot verify that if {0} then ''{1}''.",
```
Comment (jesyspa) [inline plugins/formal-verification/formver.plugin/src/org/jetbrains/kotlin/formver/FormalVerificationPluginErrorMessages.kt:60]: IMHO the quotes around {1} don't look very neat

**Review by jesyspa — APPROVED** [review]
Looks fine as a first implementation, I think there's work to do with getting the rhs user-friendly but it's not blocking.

File: plugins/formal-verification/formver.plugin/src/org/jetbrains/kotlin/formver/FormalVerificationPluginErrorMessages.kt (hunk 3 above)
Comment (gabryon99) [reply plugins/formal-verification/formver.plugin/src/org/jetbrains/kotlin/formver/FormalVerificationPluginErrorMessages.kt:60]: Ok, I'm going to remove them!

File: plugins/formal-verification/testData/diagnostics/bad_contracts/cond_effects.fir.diag.txt (hunk 2 above)
Comment (gabryon99) [reply plugins/formal-verification/testData/diagnostics/bad_contracts/cond_effects.fir.diag.txt:24]: I agree! I should check if we have a renderer for just the type name without all the rest

File: plugins/formal-verification/formver.plugin/src/org/jetbrains/kotlin/formver/reporting/VerifierErrorInterpreter.kt (hunk 1 above)
Comment (gabryon99) [reply plugins/formal-verification/formver.plugin/src/org/jetbrains/kotlin/formver/reporting/VerifierErrorInterpreter.kt:114]: You mean moving the inside `ReturnsEffect` definition?

Comment (jesyspa) [reply plugins/formal-verification/formver.plugin/src/org/jetbrains/kotlin/formver/reporting/VerifierErrorInterpreter.kt:114]: Using the same code to get messages for `ReturnsEffect` in both cases, if it's possible, so you don't have to process them in two places.  It's not blocking, though.

Comment (gabryon99) [reply plugins/formal-verification/formver.plugin/src/org/jetbrains/kotlin/formver/reporting/VerifierErrorInterpreter.kt:114]: Ah yeah! Then I have an idea.

**Pull request merged**